### PR TITLE
Dependencies guidance

### PR DESCRIPTION
**Pull request by mattmachell**
First attempt at formalising some of the approaches we already have into something a bit more solid and linkable.

Comments welcome.

### CODE REVIEW

File: source/documentation/guides/dependencies.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+You want a library that is compatible with the main framework you are using and any coding standards your team have in place. You also need to consider if it's maintaining compatibility with upcoming language and framework versions. We've seen this with Laminas and PHP 8.1 compatibilities, for example.
+
+### 3. Check that the dependency is managed by a community
+
+Single contributor libraries have more chance of being compromised and nobody noticing. More eyes on changes means more chance of vandalism or malicious activity being caught earlier. 
+
+Communities also enforce standards and tend to mean APIs are more stable as others need to use them too. Communities supported by charitable trust (JS foundation) or bit organisations (Mozilla) tend to help with ensuring maintenence, stability and backward compatibility.
```
Comment (gregtyler) [inline source/documentation/guides/dependencies.html.md.erb:42]: Little typo:
```suggestion
Communities also enforce standards and tend to mean APIs are more stable as others need to use them too. Communities supported by charitable trust (JS foundation) or big organisations (Mozilla) tend to help with ensuring maintenence, stability and backward compatibility.
```

**Comment by mattmachell** [comment]
@denialanderror that's a really good point. I'll reword to include a section on shared internal libraries and maintenance burden of internally written code too.

File: source/documentation/guides/dependencies.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+Sometimes it's not the direct dependency itself that's the risk, but the dependencies of the dependency. It's worth checking for further downstream projects with the same set of issues.
+
+### 9. Pin to a specific version
+
+It's always best to pin a given dependency to a specific version. So using 1.0.4 rather than a less specific ^1.0. 
+
+While lock files will pin to known good versions, you might inadvertantly pull in a malicious version 1.0.5 as it becomes available as part of development changes when updating packages if you aren't pinned. 
+
+We have dependabot to pull in point-version updates when they become available. For particularly volatile ecosystems it is best to wait a 3 days before merging PRs for non-security related updates to allow time for malicious activity to be detected.
```
Comment (johnnolan) [inline source/documentation/guides/dependencies.html.md.erb:82]: We are currently looking at the idea of auto merging depenabot PRs in our pipeline. See this link

https://github.com/ministryofjustice/opg-pdf-service/pull/218

I wonder is it worth exploring how to add a delay of 3 days to anything that auto merges PRs? 

Or are we safer getting merging due to any bug fixes coming out and fixing them quickly? My gut says no.

Comment (mattmachell) [reply source/documentation/guides/dependencies.html.md.erb:82]: We had auto-merging on Sirius for a bit, but turned it off after a few really bad incidents in the node ecosystem. 3 days felt like enough to catch most malicious activity. There's probably a bit more info on security vs general version bumps that I haven't made clear. So I'll do some clarifying.

**Pull request merged**